src/XRP/wallet-factory.ts

```diff
@@ -33,6 +34,38 @@ export default class WalletFactory {
     this.isTest = XrpUtils.isTestNetwork(network)
   }
 
+  /**
+   * Generate a new HD wallet with a random mnemonic and the default XRP derivation path.
+   *
+   * Secure random number generation is used when entropy is omitted and when the runtime environment has the necessary support.
+   * Otherwise, an error is thrown.
+   *
+   * Runtime environments that do not have secure random number generation should pass their own buffer of entropy.
+   *
+   * @param entropy - An optional hex string of entropy.
+   * @returns A result which contains the newly generated wallet and associated artifacts.
+   */
+  public async generateRandomHdWallet(
+    entropy: string | undefined = undefined,
+  ): Promise<HdWalletGenerationResult | undefined> {
+    if (entropy && !Utils.isHex(entropy)) {
+      return undefined
+    }
+
+    const mnemonic =
+      entropy === undefined
+        ? bip39.generateMnemonic()
```

Comment:
I thought this would throw an error when entropy is undefined?

Reply:
We prefer not to throw errors in this library because we bind it into higher level libraries and exception handling complicates that.